client/post-editor/editor-html-toolbar/index.jsx

```diff
@@ -170,11 +170,21 @@ export class EditorHtmlToolbar extends Component {
 			previousSelectionEnd + insertedContentLength;
 	}
 
-	// execCommand( 'insertText' ), needed to preserve the undo stack, does not exist in IE11.
-	// Using the previous version of replacing the entire content value instead.
-	updateEditorContent = isIE11
-		? this.updateEditorContentIE11
-		: this.insertEditorContent;
+	updateEditorContent = ( before, newContent, after ) => {
+		if ( isFirefox ) {
+			// In Firefox, execCommand( 'insertText' ), needed to preserve the undo stack,
+			// always moves the cursor to the end of the content.
+			// A workaround involving manually setting the cursor position and inserting the editor content
+			// is needed to put the cursor back to the correct position.
+			this.updateEditorContentFirefox( before, newContent, after );
+		} else if ( isIE11 ) {
+			// execCommand( 'insertText' ), needed to preserve the undo stack, does not exist in IE11.
+			// Using the previous version of replacing the entire content value instead.
+			this.updateEditorContentIE11( before, newContent, after );
+		} else {
+			this.insertEditorContent( before, newContent, after );
+		}
+	}
```

Comment:
ah. well if you are up for it (and I certainly like it) then we can leave the pattern as is with my comment…

```js
if ( isFirefox ) {
	return this.updateEditorContentFirefox;
}

if ( isIE11 ) { … }
```